src/eventHubMessageExplorer.ts

```diff
@@ -0,0 +1,99 @@
+                        .then(this.sendToEventHubDone(client, this._messageLabel))
+                        .catch(this.printError(this._outputChannel, this._messageLabel, client));
+                }
+                catch (e) {
+                    this.outputLine(this._messageLabel, e);
+                }
+            }
+        });
+    }
+
+    public startMonitorEventHubMessage(): void {
+        let eventHubConnectionString = Utility.getConfig(this._eventHubConnectionstringKey, this._eventHubConnectionStringTitle);
+        let eventHubPath = Utility.getConfig(this._eventHubPathKey, this._eventHubPathTitle);
+        if(!eventHubConnectionString || !eventHubPath) {
+            this.outputLine(this._monitorLabel, 'Invalid arguments of event hub!');
+        }
+        try {
+            this._eventHubClient = EventHubClient.fromConnectionString(eventHubConnectionString, eventHubPath);
+            let receiveAfterTime = Date.now() - 5000;
+            this._outputChannel.show();
+            this.outputLine(this._monitorLabel, 'Start monitor event hub ...');
+            this._appInsightsClient.sendEvent('EventHub.startMonitor');
+            this._eventHubClient.open()
+                .then(this._eventHubClient.getPartitionIds.bind(this._eventHubClient))
+                .then((partitionIds: EventHubClient.PartitionId[]) =>
+                    Promise.map(partitionIds, (partitionId) =>
+                        this._eventHubClient.createReceiver('$Default', partitionId, { startAfterTime: receiveAfterTime })
+                            .then((receiver: EventHubReceiver) => {
+                                receiver.on('errorReceived', this.printError(this._outputChannel, this._monitorLabel, this._eventHubClient));
+                                receiver.on('message', this.printMessage(this._outputChannel, this._monitorLabel, 'Message Received'));
+                            })))
+                .catch(this.printError(this._outputChannel, this._monitorLabel, this._eventHubClient));
+        }
+        catch (e) {
+            this.outputLine(this._monitorLabel, e);
+        }
+    }
+
+    public stopMonitorEventHubMessage(): void {
+        if(this._eventHubClient) {
+            this.outputLine(this._monitorLabel, 'Stop monitor event hub ...');
+            this._appInsightsClient.sendEvent('MonitorEventHub.stopMonitor')
+            this._eventHubClient.close();
+            this._eventHubClient = null;
+        }
+        else {
+            this.outputLine(this._monitorLabel, 'No monitor event hub job running.')
+        }
+    }
+
+    private sendToEventHub(eventBody: string, label: string): (_: EventHubSender) => Promise<void> {
+        return (sender: EventHubSender) => {
+            this.outputLine(label, 'Sending message to event hub ...');
+            return sender.send(eventBody);
+        };
+    }
+    
+    private sendToEventHubDone(client: EventHubClient, label: string) {
+        return () => {

```

Comment:
Does it have `err` parameter similar to https://github.com/formulahendry/vscode-azure-iot-toolkit/blob/master/src/messageExplorer.ts#L82  to indicate the call is failed?